### ISSUE TITLE
BUG: PoolSlot actor does not get ExposedPublisher as first message sometimes

### ISSUE THREAD

**Issue by drewhk** [akka/akka#21235]
```
java.lang.IllegalStateException: The first message must be `ExposedPublisher` but was [FromConnection(OnComplete)]
at akka.http.impl.engine.client.PoolSlot$SlotProcessor$$anonfun$waitingExposedPublisher$1.applyOrElse(PoolSlot.scala:120)
at akka.actor.Actor$class.aroundReceive(Actor.scala:484)
at akka.http.impl.engine.client.PoolSlot$SlotProcessor.akka$stream$actor$ActorSubscriber$$super$aroundReceive(PoolSlot.scala:92)
at akka.stream.actor.ActorSubscriber$class.aroundReceive(ActorSubscriber.scala:201)
at akka.http.impl.engine.client.PoolSlot$SlotProcessor.akka$stream$actor$ActorPublisher$$super$aroundReceive(PoolSlot.scala:92)
```

**Comment by 2beaucoup** [comment]
Would be fixed with the sub flow design described in https://github.com/akka/akka/issues/20570#issuecomment-242649212.

**Comment by jrudolph** [comment]
We hope that this problem is now fixed. The fix will be delivered with 2.4.10. Whoever was affected, please retry once 2.4.10 is released (due this week).

We know that there are still outstanding bugs related to pooling and pipelining (#21368) so we'll try to make incremental updates to improve the state of the client side implementation.
